sdk/schemaregistry/azure-schemaregistry-client/src/main/java/com/azure/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -0,0 +1,281 @@
+            schemaType,
+            response.getValue().getBytes(SCHEMA_REGISTRY_SERVICE_ENCODING),
+            getParseFunc(schemaType));
+
+        resetIfNeeded();
+        guidCache.put(schemaId, schemaObject);
+        logger.verbose(String.format("Cached schema object. Path: '%s'", schemaId));
+        return schemaObject;
+    }
+
+    @Override
+    public synchronized String getSchemaId(
+        String schemaGroup, String schemaName, String schemaString, String schemaType)
+        throws SchemaRegistryClientException {
+        if (schemaStringCache.containsKey(schemaString)) {
+            logger.verbose(String.format("Cache hit schema string. Group: '%s', name: '%s'", schemaGroup, schemaName));
+            return schemaStringCache.get(schemaString).getSchemaId();
+        }
+
+        SchemaId schemaId;
+        try {
+            schemaId = this.restService.getIdBySchemaContent(schemaGroup, schemaName, schemaType, schemaString);
+        } catch (HttpResponseException e) {
+            throw logger.logExceptionAsError(new SchemaRegistryClientException(
+                String.format("Failed to fetch schema guid for schema. Group: '%s', name: '%s'",
+                    schemaGroup, schemaName),
+                e));
+        }
+
+        resetIfNeeded();
+        schemaStringCache.put(
+            schemaString,
+            new SchemaRegistryObject(
+                schemaId.getId(),
+                schemaType,
+                schemaString.getBytes(SCHEMA_REGISTRY_SERVICE_ENCODING),
+                getParseFunc(schemaType)));
+        logger.verbose(String.format("Cached schema string. Group: '%s', name: '%s'", schemaGroup, schemaName));
+        return schemaId.getId();
+    }
+
+    @Override
+    public String deleteSchemaVersion(String schemaGroup, String schemaName, int version)
+        throws SchemaRegistryClientException {
+        // return this.restService.deleteSchemaVersion(schemaName, version);
+        // remove from cache
+        return null;
+    }
+
+    @Override
+    public String deleteLatestSchemaVersion(String schemaGroup, String schemaName)
+        throws SchemaRegistryClientException {
+        // return this.restService.deleteSchemaVersion(schemaName, null);
+        // remove from cache
+        return null;
+    }
+
+    @Override
+    public List<String> deleteSchema(String schemaGroup, String schemaName)
+        throws SchemaRegistryClientException {
+        // return this.restService.deleteSchema();
+        // remove from cache
+        return null;
+    }
+
+    /**
+     * Explicit call to clear all caches.
+     */
+    public synchronized void reset() {
+        guidCache.clear();
+        schemaStringCache.clear();
+        typeParserDictionary.clear();
+    }
+
+    /**
+     * Checks if caches should be reinitialized to satisfy initial configuration
+     */
+    private synchronized void resetIfNeeded() {
```

Comment:
Has a least recently used cache been thought about? Instead of clearing everything when we hit the limit just begin prune stale entries. Current logic will work but for hot path schemas they'll need to go make a network call when this case happens. 

Overall, this doesn't need to be prioritized now as 1000 entries, by default, seems to be a lot of head space.

Reply:
Yeah I thought about this but figured it wasn't worth adding.  If an app has more than 1k entries, the app is going to have a lot more problems than just the cache getting fully cleared.